Sources/NIOWebSocketServer/main.swift

```diff
@@ -203,16 +213,23 @@ let upgrader = WebSocketUpgrader(shouldUpgrade: { (head: HTTPRequestHead) in HTT
                                     channel.pipeline.add(handler: WebSocketTimeHandler())
                                  })
 
+private let httpHandler = HTTPHandler()
+
 let bootstrap = ServerBootstrap(group: group)
     // Specify backlog and enable SO_REUSEADDR for the server itself
     .serverChannelOption(ChannelOptions.backlog, value: 256)
     .serverChannelOption(ChannelOptions.socket(SocketOptionLevel(SOL_SOCKET), SO_REUSEADDR), value: 1)
 
     // Set the handlers that are applied to the accepted Channels
     .childChannelInitializer { channel in
-        let config: HTTPUpgradeConfiguration = (upgraders: [ upgrader ], completionHandler: { _ in })
+        let config: HTTPUpgradeConfiguration = (
+                        upgraders: [ upgrader ], 
+                        completionHandler: { _ in 
+                            _ = channel.pipeline.remove(handler: httpHandler)
+                        }
+                    )
         return channel.pipeline.configureHTTPServerPipeline(withServerUpgrade: config).then {
-            channel.pipeline.add(handler: HTTPHandler())
+            channel.pipeline.add(handler: httpHandler)
```

Comment:
I think this is not correct as this way you will share the same `httpHandler` over multiple connections. I think this should be `HTTPHandler()`

Reply:
I moved the definition of the httpHandler

Reply:
@woodcrust that works....